lib/web/resources.go

```diff
@@ -212,7 +214,66 @@ func (h *Handler) getGithubConnectorsHandle(w http.ResponseWriter, r *http.Reque
 		return nil, trace.Wrap(err)
 	}
 
-	return getGithubConnectors(r.Context(), clt)
+	connectors, err := getGithubConnectors(r.Context(), clt)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	defaultConnectorName, defaultConnectorType, err := ProcessDefaultConnector(r.Context(), clt, connectors)
```

Comment:
Since `ProcessDefaultConnector` updates auth preference, IMHO, its not wise to write to the backend in a `GetX` request.

Reply:
I see where you're coming from, the issue however is that the alternative to this is that what is stored in the actual backend auth preference will be out of sync with what the client sees. The alternative would be to have a separate endpoint that handles this and then chain them together, which would be overly complex for what is essentially just an issue with naming.